### ISSUE TITLE
Building windows agent fails with libwazuhext.dll error

### ISSUE THREAD

**Issue by truncet** [wazuh/wazuh#11642]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2.5 | Installation | Agent | Sources | Windows |

Hi , I am trying to install windows agent from sources.
I am getting below error.

CC libwazuhext.dll
external/sqlite/libsqlite3.a(sqlite3.o):sqlite3.c:(.text+0x35cb8): undefined reference to `__divmoddi4'
collect2: error: ld returned 1 exit status
Makefile:1499: recipe for target 'libwazuhext.dll' failed
make[1]: *** [libwazuhext.dll] Error 1
make[1]: Leaving directory '/root/wazuh-4.2.5/src'
Makefile:746: recipe for target 'winagent' failed
make: *** [winagent] Error 2

To Reproduce:-
curl -Ls https://github.com/wazuh/wazuh/archive/v4.2.5.tar.gz | tar zx
cd wazuh-4.2.5/src

make clean-deps
make clean

make deps TARGET=winagent
make TARGET=winagent

**Comment by jotacarma90** [comment]
Hi @truncet,

I have tried to reproduce your error with the same steps, but the compilation finishes correctly.
I've been researching through Wazuh's community history, and no one has ever reported a similar problem, so I've started researching outside of Wazuh. It looks like the function you are missing:

`undefined reference to __divmoddi4`

Is related to gcc compiler version or environment architecture, I found this:
_"This function is used to divide two 64-bit integers on i686, but not on x86_64 because of hardware support for 64-bit division. It is available in libgcc unless we are building the CRT (winpthreads is also sufferring from this problem). I am not familiar with cygwin, but despite that I think mingw-w64 could provide it."_

Link:
https://gcc.gnu.org/legacy-ml/gcc-help/2017-05/msg00108.html

This leads me to think that maybe you are missing some of the dependencies to install or update some of them. In the docu we go over how to install them all, also make sure you have the whole system up to date:
`apt-get install gcc-mingw-w64 g++-mingw-w64-i686 g++-mingw-w64-x86-64 nsis make cmake`
`apt-get update`

Link to the docu:
https://documentation.wazuh.com/current/installation-guide/more-installation-alternatives/wazuh-from-sources/wazuh-agent/index.html

If the problem continues, let me know and we will investigate further, I hope this has been helpful.
Regards

**Comment by truncet** [comment]
Hi @jotacarma90,
 I had installed the dependencies as mentioned in documentation. 
I tried it again with fresh new install and the issue still persists.
My os is:-
NAME="Ubuntu"
VERSION="16.04.1 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.1 LTS"
VERSION_ID="16.04"

Regards

**Comment by vikman90** [comment]
@truncet @jotacarma90  I managed to reproduce this issue on Ubuntu 16.04:

```sh
$ make TARGET=winagent
    CC libwazuhext.dll
external/sqlite/libsqlite3.a(sqlite3.o):sqlite3.c:(.text+0x35cb8): undefined reference to `__divmoddi4'
collect2: error: ld returned 1 exit status
```

That happens because the SQLite3 dependency was built on a newer version of MinGW. We can force to build the library, but we will get into more troubles:

```sh
$ rm external/sqlite/libsqlite3.a external/sqlite/sqlite3.o
$ make TARGET=winagent
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.12.4 or higher is required.  You are running version 3.10.2
```

We need CMake 3.12.4+, we can get it and build it from sources, as explained at [Installing Wazuh agent from sources](https://documentation.wazuh.com/current/installation-guide/more-installation-alternatives/wazuh-from-sources/wazuh-agent/index.html):

```sh
apt-get install g++ libssl-dev
curl -OL https://packages.wazuh.com/utils/cmake/cmake-3.18.3.tar.gz && tar -zxf cmake-3.18.3.tar.gz
cd cmake-3.18.3 && ./bootstrap --no-system-curl
make -j$(nproc) && make install
cd .. && rm -rf cmake-*
```

However, the compilation will fail again:
```sh
$ make TARGET=winagent
[ 14%] Building CXX object CMakeFiles/sysinfo.dir/src/network/networkInterfaceWindows.cpp.obj
In file included from /root/wazuh-4.2.5/src/data_provider/src/network/networkInterfaceWindows.cpp:13:0:
/root/wazuh-4.2.5/src/shared_modules/utils/windowsHelper.h:25:28: fatal error: versionhelpers.h: No such file or directory
```

That's because `versionhelpers.h` is not available in MinGW 5. We require MinGW 7+ as of Wazuh v4.2.

So, I suggest the following fixes:

- Use Ubuntu 18.04+, and install CMake 3.18 as shown above.
- Use Ubuntu 20.04+, where the steps you followed should work.

Since we are using Linux to cross-compile the agent for Windows, I hope you can use a modern Ubuntu version. However, this problem tells us that we need to update the Windows build procedure documentation.

Thank you for notifying this issue.
Best regards.
